File: frontend/libs/reporting-api/src/lib/model/create-run-case-payload.ts
import { Annotation } from './annotation';
import { TestLocation } from './test-location';
import { TestStatus } from './test-status';

export interface CreateRunCasePayload {
  id: string;
  title: string;
  location?: TestLocation;
  tags?: string[];
  timeout?: number;
  retries?: number;
  annotations?: Annotation[];
  expectedStatus?: TestStatus;
}

File: backend/OpenWright.Api/Reporting/Payloads/CreateRunCasePayload.cs
using OpenWright.Api.Projects.ValueObjects;
using OpenWright.Api.Runs.ValueObjects;

namespace OpenWright.Api.Reporting.Payloads;

public class CreateRunCasePayload
{
    public required Guid Id { get; init; }
    public required string Title { get; init; }
    public string[]? Tags { get; init; }
    public decimal? Timeout { get; init; }
    public int? Retries { get; init; }
    public Annotation[]? Annotations { get; init; }
    public TestStatus? ExpectedStatus { get; init; }
}
